src/lib.rs

```diff
@@ -9,3 +13,55 @@ pub mod good_fences_runner;
 pub mod import_resolver;
 mod path_utils;
 pub mod walk_dirs;
+
+#[napi]
+pub fn good_fences(
+    paths: Vec<String>,
+    project: String,
+    base_url: Option<String>,
+    root: String,
+    output: Option<String>,
+) {
+    let args = cli::Cli {
+        paths,
+        project,
+        base_url,
+        root,
+        output: output.unwrap_or("good-fences-violations.json".to_owned()),
+    };
```

Comment:
I don't think it's a good idea to have the api wrap the cli, it should be the other way around where the CLI and the api surface both consume the same package epxposed to node

Also, the shared logic should just return the struct. The cli can serialize and write it if you want, but the API should definitely not write to disk.

Reply:
It should be straightforward to return the violations json as a javascript object with serde, with the [serde-json crate feature on napi](https://docs.rs/napi/latest/napi/#serde-json)

Reply:
Seems that this is the current good-fences interface that owa-build uses
```
export default interface GoodFencesError {
    message: string;
    sourceFile?: string;
    rawImport?: string;
    fencePath?: string;
    detailedMessage: string;
}
```
Since the `ImportRuleViolation` borrows values from `GoodFencesRunner`, ended up adding a struct `GoodFencesError` that only converts those &str to it's owned values. Suggestions are welcome on how to better handle this, if you find the need to still emmit `ImportRuleViolation`  as is to JS I thinkg doing it in a PR before this one will be beter.